sycl/include/sycl/builtins.hpp

```diff
@@ -734,8 +734,8 @@ std::enable_if_t<__FAST_MATH_GENFLOAT(T), T> sin(T x) __NOEXC {
 
 // svgenfloat sincos (svgenfloat x, genfloatptr cosval)
 template <typename T, typename T2>
-std::enable_if_t<
-    detail::is_svgenfloat<T>::value && detail::is_genfloatptr<T2>::value, T>
+std::enable_if_t<__FAST_MATH_GENFLOAT(T) && detail::is_genfloatptr<T2>::value,
```

Comment:
I'm not sure where `__FAST_MATH_WITH_GENFLOAT` comes from or what it is, but is there any chance that these two enable_if might both be true at the same time and then result in an ambiguous call?   Do you need to ensure they are mutuallly exclusive?